code/obj/misc_junk.dm

```diff
@@ -86,24 +86,32 @@
 			last_laugh = world.time
 
 	process()
-		if(prob(50) || current_state < GAME_STATE_PLAYING) // Takes around 12 seconds for ol chompski to vanish
+		if (prob(50) || current_state < GAME_STATE_PLAYING) // Takes around 12 seconds for ol chompski to vanish
 			return
-		// No teleporting if youre in a crate
-		if(istype(src.loc,/obj/storage) || istype(src.loc,/mob/living))
+		// No teleporting if youre in a container
+		if (istype(src.loc,/obj/storage) || istype(src.loc,/mob/living))
 			return
 		// Nobody can ever see Chompski move
-		for(var/mob/M in viewers(src))
-			if(M.mind) // Only players. Monkeys and NPCs are fine. Chompski trusts them.
+		for (var/mob/M in viewers(src))
+			if (M.mind) // Only players. Monkeys and NPCs are fine. Chompski trusts them.
 				return
 		//oh boy time to move
-		playsound(src.loc,"sound/misc/gnomechuckle.ogg" ,50,1)
-		var/obj/crate = pick(by_type[/obj/storage])
-		while(crate.z != 1)
-			crate = pick(by_type[/obj/storage])
-		src.set_loc(crate)
 
+		var/list/potential_containers = by_type[/obj/storage].Copy()
+
+		var/obj/storage/container = null
+		while (length(potential_containers))
+			var/obj/storage/random_container = pick(potential_containers)
+			if (random_container.open == 0 && random_container.z == 1)  // container is closed and on station z-level
+				container = random_container
+				break
+			potential_containers -= container
```

Comment:
This can be potentially slow. Possible fixes:
a.) swap `random_container` with the last element, then remove the last element (which is fast)
b.) don't use `pick`, instead pre-shuffle the list at the beginning using the `shuffle_list` macro, then iterate over containers in order
c.) don't fix anything and keep it like this because it's unlikely to be a performance concern anyway
d.) revert back to the pre-PR code which used just repeated `pick`s which can *potentially* take longer time than all this but on average will be faster because `.Copy()` needs to copy the whole list which isn't fast either

Reply:
Optimized per suggestions in an internal discussion.